egs/lrs/asr1/run.sh

```diff
@@ -0,0 +1,369 @@
+datadir=		     # The LRS2 dataset directory e.g. /home/foo/LRS2
+
+pretrain=true		     # if use LRS2 pretrain set 
+segment=true  		     # if do segmentation for pretrain set
+
+# bpemode (unigram or bpe)
+nbpe=500
+bpemode=unigram
+
+## train_lm=false, we have to download pretrained Librispeech language model from 
+## https://drive.google.com/open?id=1BtQvAnsFvVi-dp_qsaFP7n4A_5cwnlR6
+function gdrive_download () {
+  CONFIRM=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate "https://docs.google.com/uc?export=download&id=$1" -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')
+  wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$CONFIRM&id=$1" -O $2
+  rm -rf /tmp/cookies.txt
+}
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+# define sets
+if [ "$pretrain" = true ] ; then
+	train_set="pretrain_Train"
+else
+	train_set="Train"
+fi
+train_dev="Val"
+recog_set="Val Test"
+
+# Stage -1: Data Download
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    if [ -d "$datadir" ]; then
+    	echo "Dataset already exists."
+    else
+    	echo "For downloading the data, please visit 'https://www.robots.ox.ac.uk/~vgg/data/lip_reading/lrs2.html'."
+    	echo "You will need to sign a Data Sharing agreement with BBC Research & Development before getting access."
+    	echo "Please download the dataset by yourself and save the dataset directory in path.sh file"
+    	echo "Thanks!"
+    fi
+fi
+
+# Stage 0: Data preparation
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    echo "stage 0: Data preparation"
+    for part in Test Val Train; do 
+        local/data_preparation.sh $datadir $part $segment $nj || exit 1;
+    done
+    if [ "$pretrain" = true ] ; then
+    	part=pretrain
+    	local/data_preparation.sh $datadir $part $segment $nj || exit 1;
+    fi
+    for part in pretrain Test Val Train; do 
+    	mv data/${part} data/${part}_org || exit 1;
+    done
+    echo "stage 0: Data preparation finished"
+
+fi
+
+feat_tr_dir=${dumpdir}/${train_set}/delta${do_delta}; mkdir -p ${feat_tr_dir}
+feat_test_dir=${dumpdir}/Test/delta${do_delta}; mkdir -p ${feat_test_dir}
+feat_val_dir=${dumpdir}/Val/delta${do_delta}; mkdir -p ${feat_val_dir}
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    ### Task dependent. You have to design training and dev sets by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 1: Feature Generation"
+
+    # remove utt having more than 3000 frames
+    # remove utt having more than 400 characters
+    remove_longshortdata.sh --maxframes 3000 --maxchars 400 data/Train_org data/Train
+    remove_longshortdata.sh --maxframes 3000 --maxchars 400 data/Test_org data/Test
```

Comment:
Is this a standard way of removing the utterances in test and validation set? Normally, we will keep all the samples in test and validation for fair comparison with other systems.

Reply:
yes you are right, we have segmented the pretrain set, so this code is not necessary